### PR TITLE
Upgrade to Flutter 1.22 UI

### PR DESCRIPTION
**Pull request by theswerd**
### Upgrade to Flutter 1.22

- Upgraded to **Flutter 1.22**
- Replaced all **flatButtons** with their replacement **textButtons**
- Incorporated **Padding** in **PageButtons** into the button styling
- Started Restyling code to new Flutter standards -- still more to do

##### Screenshots:
<details><summary><b>Old Screenshots</b></summary>

Simulator Screen Shot - iPhone SE (2nd generation) - 2020-10-05 at 14 43 25

Simulator Screen Shot - iPhone SE (2nd generation) - 2020-10-05 at 14 43 02

Simulator Screen Shot - iPhone SE (2nd generation) - 2020-10-05 at 14 38 25

</details>

<details><summary><b>New  Screenshots</b></summary>

Simulator Screen Shot - iPhone SE (2nd generation) - 2020-10-05 at 16 07 54

Simulator Screen Shot - iPhone SE (2nd generation) - 2020-10-05 at 16 07 46

Simulator Screen Shot - iPhone SE (2nd generation) - 2020-10-05 at 16 07 30
</details>

#### How did you test the change?
* [x] iOS Simulator
* [ ] Android Emulator
* [ ] iOS Device
* [ ] Android Device
* [ ] `curl` to a dev App Engine server

---

#### Please follow this checklist. Please check each appropriate box (put an 'x' or check it after creating the PR).

- [ ] REQUIRED: Do you have an Issue **assigned to you** for this PR?
- [x] Provided detailed pull request description and a succinct title
- [x] Tested your changes, especially after any code review iterations.
- [x] Included any relevant screenshots of UI updates.
- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md).
- [x] Verified all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE) file in the root of the repository.
- [x] Verified your name is in the [docs/credits.yaml](https://github.com/WorldHealthOrganization/app/blob/master/docs/credits.yaml) file (if you want it to be).

### CODE REVIEW

**Review by brunobowden — CHANGES_REQUESTED** [review]
@theswerd - this is a welcome improvement. Couple of overall thoughts:

Need a Flutter expert to look at this which isn't me. Maybe @britannio or pinging on the #flutter channel on Slack would work for this.

`-nullsafety` is using prerelease libraries. This should be split out to another commit and a separate discussion. At this point, I don't think that the libraries are meant for public discussion and it's not worth the risk. It's a big decision to use prerelease libraries, so it's essential that important like that should be in the commit description! Given this fix, I thought you wouldn't see this as a suggestion:
https://github.com/dart-lang/sdk/issues/42657

Screenshots should include the main page as that's the most important one. I've scoped your screenshots in <details>...</details> as otherwise the layout becomes unmanageable in GitHub.

If you think that a series of improvements is need for Flutter 1.22. Please create issues for them. Maybe even a combined issue with several items listed out. It's important to do modest incremental changes, so that if there's an issue, we can better bisect to find the cause.

File: client/flutter/pubspec.lock [hunk 1]
```diff
@@ -797,5 +797,5 @@ packages:
     source: hosted
     version: "2.2.0"
 sdks:
-  dart: ">=2.9.1 <3.0.0"
-  flutter: ">=1.20.2 <2.0.0"
+  dart: ">=2.10.0-110 <2.11.0"
```
[the remark this answers is not in the document]
Comment (britannio) [reply client/flutter/pubspec.lock:800]: I don't think this is an issue.

File: client/flutter/lib/components/dialogs.dart [hunk 2]
```diff
@@ -18,9 +18,10 @@ class Dialogs {
           title: Text(title),
           content: body ?? Text(bodyText),
           actions: <Widget>[
-            FlatButton(
-              // TODO: Localize
-              child: Text(S.of(context).commonDialogButtonOk),
+            TextButton(
+              child: Text(
```
[the remark this answers is not in the document]
Comment (theswerd) [reply client/flutter/lib/components/dialogs.dart:22]: I believe that TODO is outdated, as there are no hard coded strings in this file

File: client/flutter/lib/components/page_button.dart [hunk 3]
```diff
@@ -61,51 +61,67 @@ class _PageButtonState extends State<PageButton> {
...
-      color: widget.backgroundColor,
-      child: Padding(
-        padding: EdgeInsets.symmetric(
-          vertical: this.widget.verticalPadding,
-          horizontal: this.widget.horizontalPadding,
+
+      // disabledColor: Constants.neutralTextLightColor.withOpacity(0.4),
+      // disabledTextColor: Colors.white,
+      //
+      // color: widget.backgroundColor,
```
[the remark this answers is not in the document]
Comment (theswerd) [reply client/flutter/lib/components/page_button.dart:70]: Ah Yep -- Will delete that today

**Comment by brunobowden** [comment]
Need resolution on Flutter 1.22.0 `-nullsafety` dependencies: https://github.com/flutter/flutter/issues/67473

**Review by yukuairoy — APPROVED** [review]

**Comment by brunobowden** [comment]
@theswerd - time to rebase on master and see what's left of the commit. I still think there's some worthwhile changes in here.

**Pull request closed without merging**